### DISCUSSION TITLE
Handling error when `fallback: true` is enabled

### DISCUSSION THREAD

**Discussion by zackkrida** [vercel/next.js#11862] [Help] [answered]
Hi there! If I'm using `getStaticProps` and `getStaticPaths` with the `fallback: true` setting and a page _actually doesn't exist_ when the fallback tries to run `getStaticProps`, what would be the best way to capture that error?

This is the error I'm referring to: https://github.com/zeit/next.js/blob/b274fe39d38111588e625342a89dc568ad38fff5/packages/next/next-server/lib/router/router.ts#L110

~~I'm guessing `Router.events.on("routeChangeError", ...)` is the way to go but wanted to confirm first. Thank you!~~

**Comment by Timer** [comment, marked as answer]
Your `getStaticProps` should return a boolean like `notFound: true`, and your page can render a 404 page or the `next/error` page. This would allow the page to eventually become real (if the missing slug is created in backend).

**Reply by zackkrida** [reply]
Ah, i didn't even think to handle it through the `getStaticProps` response, great idea. Thanks!

**Reply by timneutkens** [reply]
@zackkrida I posted a full reply on how to do it here: https://github.com/zeit/next.js/discussions/10960#discussioncomment-1201

**Reply by insivika** [reply]
Unfortunately this does not work for me. I get the react error modal with ` throw new Error('Failed to load static props')` in development and `An unexpected error has occurred.` in production

**Reply by aomini** [reply]
@insivika did you manage to fix that? I'm having the same issue cannot use try catch in `getStaticProps` and getStaticProps throws unhandled error.

**Reply by insivika** [reply]
Yeah my API call was returning `undefined` when trying to build the fallback so I put an if clause in there to set the prop to null if the api result is `undefined`. At that point [this example](https://github.com/vercel/next.js/discussions/10960#discussioncomment-1201) will work just fine

**Reply by SalahAdDin** [reply]
What's the solution for this problem?

**Reply by mayank1513** [reply]
The error occurs when some exception is thrown. So best way is to use try catch in getStaticProps to handle fallback paths.

**Reply by mayank1513** [reply]
> 
> 
> Unfortunately this does not work for me. I get the react error modal with ` throw new Error('Failed to load static props')` in development and `An unexpected error has occurred.` in production

Make sure there are no exceptions throws - checkout this answer  - https://stackoverflow.com/a/67946220/9640177

**Comment by mayank1513** [comment]
Checkout this answer

https://stackoverflow.com/a/67946220/9640177
